docs/src/main/asciidoc/resource-handling.adoc

```diff
@@ -103,9 +109,66 @@ spring:
           endpoint:  ${STORAGE_FILESHARE_ENDPOINT}
 ----
 
+==== Autowire a resource
+You can use the annotation of `@Value("azure-blob://[your-container-name]/[your-blob-name]")` to autowire a blob Resource.
+----
+@Value("azure-blob://[your-container-name]/[your-blob-name]")
+private Resource storageResource;
+----
 
-=== Samples
+You can use the annotation of @Value("azure-file://[your-fileshare-name]/[your-file-name]") to autowire a file Resource.
+
+----
+@Value("azure-file://[your-fileshare-name]/[your-file-name]")
+private Resource storageResource;
+----
+
+==== Read and write to a resource
+
+You can read a resource from Azure Blob storage with `getInputStream()` method.
+
+----
+ this.storageResource.getInputStream();
+----
+
+You can write to a resource in Azure Blob storage by casting the Spring `Resource` to `WritableResource`.
+
+----
+ ((WritableResource) this.azureBlobResource).getOutputStream();
+----
+
+==== Other operations 
+The Spring Resource abstraction for Azure Storage also supports link:https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#resources[other operations] defined in Spring's `Resource` and `WritableResource` interface. 
+
+==== Autowire the BlobServiceClientBuilder
+You can autowire the `BlobServiceClientBuilder` and create a client using:
+----
+@Autowire
+private BlobServiceClientBuilder blobServiceClientBuilder;
+
+private final BlobServiceAsyncClient blobServiceAsyncClient = blobServiceClientBuilder.buildAsyncClient();
+
+----
```

Comment:
If we want to mention the direct usage of Azure SDK clients here, we should put it behind all samples related to Resource APIs.